pkg/cluster/cluster.go

```diff
@@ -211,11 +211,10 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 		if needDeleteCluster {
 			c.logger.Infof("deleting cluster")
 			c.delete()
+			go c.reportFailedStatus()
```

Comment:
I am going to fix this in another pr. needDeleteCluster is incorrect and confusing at the first place. That is why we put c.reportFailedStatus() outside before without even noticing it is wrong.